cmd/keymaster/main.go

```diff
@@ -745,10 +744,17 @@ func main() {
 		logger.Fatal(err)
 	}
 
-	signer, _, err := genKeyPair(privateKeyPath)
+	tempPrivateKeyPath := filepath.Join(homeDir, DefaultKeysLocation, "keymaster-temp")
+	signer, tempPublicKeyPath, err := genKeyPair(tempPrivateKeyPath, userName+"@keymaster")
+	if err != nil {
```

Comment:
Please add a `defer os.Remove(tempPrivateKeyPath)` to ensure the temporary file is deleted on failure. Same comment applies for other temporary files.

Reply:
What about the defer()?